test/e2e/mock_schemas.py

```diff
@@ -132,3 +133,13 @@
     "code": "system-c",
 }
 # pylint: enable=duplicate-code
+
+# Load in the usage statuses data
+with open("./data/usage_statuses.json", "r", encoding="utf-8") as file:
+    usage_statuses_data = json.load(file)
+    USAGE_STATUSES = [
+        {"_id": usage_status["_id"]["$oid"], "value": usage_status["value"]} for usage_status in usage_statuses_data
+    ]
+USAGE_STATUSES_EXPECTED = [
+    {"id": usage_status["_id"], "value": usage_status["value"]} for usage_status in USAGE_STATUSES
+]
```

Comment:
Do we have to load the usage statuses from the JSON file? Can't we just hardcode a couple of usage statuses (they can even be dummy ones)? I don't think we need to insert all the usage statuses for the purposes of testing anyway.

Reply:
We do the same for units, I agree it doesn't necessarily need all of them but its nice as a check that we haven't broken the syntax of the file. With full crud, this can be tested like the others. We also have the possibility of pre-populating the test data with the units, usage_statuses and predefined catalogue item property using the dev script in the CI if we wanted to (Which is what I have done on the front end api tests), although the expected values would still have to be loaded.

Reply:
I see. Given that we are now doing checks in the API, it means that it now cares what usage status you give it so this functionality now makes sense to stay in the API and with this in mind, I believe that we should support full CRUD (maybe not update and delete to start with as it would require migrations). You mentioned that Joshua wanted to do the full CRUD for the property templates but I see this functionality here as more important. We should do the same for units.

Reply:
> (maybe not update and delete to start with as it would require migrations)


I'm guessing the patch and delete should work like manufacturers. A user/admin should not be able to delete a usage status if the usage status is being used in an item, and the patch would work if we used IDs instead of the value, but if we use the value, it would be the same as the delete.

Reply:
> I see. Given that we are now doing checks in the API, it means that it now cares what usage status you give it so this functionality now makes sense to stay in the API and with this in mind, I believe that we should support full CRUD (maybe not update and delete to start with as it would require migrations). You mentioned that Joshua wanted to do the full CRUD for the property templates but I see this functionality here as more important. We should do the same for units.

Yeah that sounds sensible to me.



> > (maybe not update and delete to start with as it would require migrations)
> 
> I'm guessing the patch and delete should work like manufacturers. A user/admin should not be able to delete a usage status if the usage status is being used in an item, and the patch would work if we used IDs instead of the value, but if we use the value, it would be the same as the delete.

I think as there is no way to really find those items that use that usage status and it certainly doesn't feel very feasible to expect the user to change them if there are a lot, it does make more sense to me that the edit for an admin functionality like that would require a migration rather than a normal patch as Viktor suggested. Units certainly would pose another challenge in the way that they are stored if we were to try and look up whether they are used or not. Again I think a migration is the only way if they wanted to change the standard unit they use for something for example although I would hope no unit just ceases to exist.